lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts

```diff
@@ -152,20 +172,34 @@ export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent i
     }
 
     displayMenuOption(option: string): boolean {
-        return this.field.params.menuOptions ? this.field.params.menuOptions[option] : option !== 'retrieveMetadata';
+        return this.field?.params?.menuOptions ? this.field.params.menuOptions[option] : option !== AttachFileCloudWidgetComponent.RETRIEVE_METADATA_OPTION;
+    }
+
+    onRowClicked(file?: Node) {
+        if (this.selectedNode?.id === file?.id) {
+            this.selectedNode = null;
+        } else {
+            this.selectedNode = file;
+        }
+        this.contentModelFormFileHandler(this.selectedNode);
     }
 
-    onRetrieveFileMetadata(file: Node) {
-        const values: FormValues = {};
-        const metadata = file?.properties;
-        if (metadata) {
-            const keys = Object.keys(metadata);
-            keys.forEach(key => {
-                const sanitizedKey = key.replace(':', '_');
-                values[sanitizedKey] = metadata[key];
+    contentModelFormFileHandler(file?: Node) {
+        if (file?.id && this.field?.params?.menuOptions && this.field.params.menuOptions[AttachFileCloudWidgetComponent.RETRIEVE_METADATA_OPTION]) {
```

Comment:
can we have this condition as a function to make the code easier to read? `isFileWhatever(file)`

Reply:
sure 👍